Controllers/ReadersController.cs

```diff
@@ -95,13 +103,72 @@ public async Task<IActionResult> Borrow([FromRoute] int readerId, [FromRoute] in
         {
             TempData["AlertMessage"] = "Reader has already borrowed 5 books that are not yet returned.";
             TempData["AlertType"] = "warning";
-            return RedirectToAction(nameof(GetReadersFromDb));
+        }
+            return RedirectToAction(nameof(GetPaginatedReadersFromDb));
+
+    }
+    [HttpPost]
+    public async Task<IActionResult> BorrowBook(int readerId, int bookId, string? cancel)
+    {
+        if (!string.IsNullOrEmpty(cancel))
+        {
+            return RedirectToAction(nameof(GetPaginatedReadersFromDb));
+        }
+
+        if (await _readerRepository.HasReachedBorrowLimitAsync(readerId))
```

Comment:
this check is already being done inside : await _readerRepository.AddReaderBookAsync(readerId, bookId);